### PR TITLE
Accept json

### LINKED ISSUE #1383: HTTP handlers need to all check Accept: headers
### Description
Some of our HTTP handlers just return JSON with nary a care in the world about what the requester actually wanted.

In case we want to support multiple protobuf or other encodings for these endpoints in the future, we should explicitly deny requests that specify anything other than JSON.

### Success criteria (What criteria will consider this ticket closeable?)
When asked explicitly for a certain encoding, none of PIlosa's handlers should respond with anything other than the encoding requested, or an error.

### PR DESCRIPTION
**Pull request by tgruben**
## Overview
Adds check for Accept headers to confirm clients accepts application/json responses.  Note this might require a fix to all the clients @yuce .   Just make sure they contain the "Accept:application/json" header for those requests that you are expecting json

Fixes #1383

### CODE REVIEW

File: test/pilosa_test.go [hunk 1]
```diff
@@ -32,12 +33,21 @@ func TestNewCluster(t *testing.T) {
...
+		strings.NewReader(""),
+	)
 
-	response, err := http.Get("http://" + cluster[0].Server.Addr().String() + "/status")
+	req.Header.Set("Accept", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
-		t.Fatalf("getting schema: %v", err)
+		panic(err)
```
Comment (jaffee) [inline test/pilosa_test.go]: why panic?

File: test/pilosa.go [hunk 2]
```diff
@@ -263,10 +263,15 @@ func (m *Main) RecalculateCaches() error {
...
-	if err != nil {
-		panic(err)
-	}
+	req, err := gohttp.NewRequest(
+		method,
+		urlStr,
+		strings.NewReader(body),
+	)
+
+	req.Header.Set("Content-Type", "application/json")
```
Comment (jaffee) [inline test/pilosa.go]: is MustDo always used for json requests? maybe we should not set headers in here at all.

Comment (tgruben) [reply test/pilosa.go]: you don't have access to headers outside of the `MustDo` function.  its not that big a deal now that the Accept is not required.  I can remove if you like, i don't see any value either way

File: http/handler.go [hunk 3]
```diff
@@ -257,8 +258,31 @@ func (h *Handler) handleHome(w http.ResponseWriter, r *http.Request) {
...
+	sendError := false
+	if found {
+		sendError = true
+		for _, v := range v {
+			if v == "application/json" {
+				sendError = false
+
+			}
+		}
+
```
Comment (jaffee) [inline http/handler.go]: some needless xtra whitespace in this func

File: test/pilosa.go (hunk 2 above)
Comment (jaffee) [reply test/pilosa.go]: all good

**Pull request merged**